packages/vm/src/errors/vm_error.rs

```diff
@@ -86,6 +86,11 @@ pub enum VmError {
     WriteAccessDenied { backtrace: BT },
     #[error("Maximum call depth exceeded.")]
     MaxCallDepthExceeded { backtrace: BT },
+    #[error(
+        "The called function args arity mismatch, expected {} args.",
+        actual_arity
+    )]
+    WrongFunctionArity { actual_arity: usize, backtrace: BT },
```

Comment:
This error type somehow captures two different error case, which makes naming and auditing hard:
1. The `args` array contains a wrong number of elements
2. The function exported by the contract is malformed

I think it would be better to check those things separately

Reply:
From the VM point of view the difference in arity could be caused by `wasmd` (for e.g. version mismatch of `wasmd` and `wasmvm` on the chain) or could be caused by contract developer (though there are other arity checks of the entry points along the way). So VM does not have any information other than the mismatch occurrence. I guess we can play with the name of the error - maybe `FunctionArityMismatch`? Only `wasmd` can reframe the meaning of this error into something more usable

Reply:
Right, sounds good. Looking at `pub fn call_migrate_raw` in this repo we see that the number of arguments is pretty much the correct/expected arity. So there is no real point in adding the expected number of args as a separate value. I like `FunctionArityMismatch`.

Reply:
Done